src/components/student/Edit.js

```diff
@@ -21,6 +24,49 @@ const useStyles = makeStyles({
 
 const Edit = () => {
   const classes = useStyles();
+  const { id } = useParams();
+  const history = useHistory();
+  const [student, setStudent] = useState({
+    stuname: "",
+    email: "",
+  });
+
+  useEffect(() => {
+    async function getStudent() {
+      try {
+        const student = await axios.get(`http://localhost:3333/students/${id}`);
+        console.log(student.data);
+        setStudent(student.data);
+        // console.log("id", student.id);
+        // console.log("name", student.stuname);
+      } catch (error) {
+        console.log("Something is Wrong");
+      }
+    }
+    getStudent();
+  }, [id]);
+
+  function onTextFieldChange(e) {
+    setStudent({
+      ...student,
+      [e.target.name]: e.target.value,
+    });
+  }
+
+  async function onFormSubmit(e) {
+    e.preventDefault();
+    try {
+      await axios.put(`http://localhost:3333/students/${id}`, student);
```

Comment:
Don't hardcode the url
Keep it in a seperate file as config.ts
And put all the endpoints there in an object and export accordingly wherever you need